Added new design for ND explore bar

DmitryJakubovsky opened the pull request:
## Description of the change

* Added new design
* Refactored ToggleBar

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #378

## Checklists

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

kbinney commented:
Nice! Thanks Dmitry.

A couple of tweaks (sorry this wasn't in a mock you could access info from, hopefully this helps):

1. The time period filter should be second from the left on the bar. So the first filter selection after the Count/rate grouping
2. Echoing Josh, for community-> explore, the location filter should be titled "Office" (for the options including Bismarck, Mandan, etc). For facilities -> explore, the location filter should be titled "County of Residence" (for the options including Adams county, barnes county, etc)
3. The filter bar should not have any space between the left side bar and it, nor between the header bar and it. The vertical space between the filter bar and the important notes bar should be the space as the vertical space between the important notes bar and the graphs. 
3. I have some tweaks to font colors, sizes, etc. I grabbed the CSS attributes for each component from the design file (unfortunately not figma), so here goes: 

- Count & rate words: 
opacity: 0.8;
font-family: Roboto-Medium;
font-size: 14px;
color: #7C8695;
letter-spacing: 0.79px;
line-height: 17px;

- selection titles (Time period, supervision type, office / county of residence)
font-family: Roboto-Medium;
font-size: 16px;
color: #72777A;
letter-spacing: 0.49px;
line-height: 21px;

- Selection options (both the visible option - All, etc -  and the full set of options that appears when you click in - only probation, only parole, etc) 
opacity: 0.8;
font-family: Roboto-Medium;
font-size: 14px;
color: #7C8695;
letter-spacing: 0.79px;
line-height: 17px;

This is what it should look like for Community -> explore, note it always touching the top and side and the time period selection to left of supervision type selection:
![slim copy](https://user-images.githubusercontent.com/19810267/84327147-9f7f3680-ab4c-11ea-8dca-5000c5bddc57.png)

This is what it should look like for Facilities -> explore, note the time period selection to the left of the county of residence selection and the title of that selection.
![slim copy 2](https://user-images.githubusercontent.com/19810267/84327380-2df3b800-ab4d-11ea-9618-af5cb9f85109.png)


No need for shrink-on-scroll behavior and the reactive behavior when the page gets narrower works for me. 

Thanks!

DmitryJakubovsky commented on src/components/toggles/ToggleBar.js:
```diff
@@ -15,59 +15,153 @@
-              />
+              {setChartDistrict && (
+                <div className="toggle-filters__filter" id="districtToggle">
+                  <span className="toggle-filters__filter-title">Office</span>
```
Renamed titles. `All` options were changed to universal "All" like in mockups bellow. As I understood it does not have a big matter.

DmitryJakubovsky commented on src/components/toggles/helpers.js:
```diff
@@ -0,0 +1,95 @@
+    ? collectDistrictOfficesOptions(districtOffices)
+    : collectDistrictsOptions(districts);
+
+export const appendAllOption = (options) => [defaultDistrictOption, ...options];
```
Thanks for your remark!

jessex commented on public/index.html:
```diff
@@ -1,82 +1,81 @@
-</html>
+</body>
 
+</html>
```
We need to add this newline back to the end of the file.

But also, why did we dedent everything in this file? Did a linter run over it? Just curious.

DmitryJakubovsky replied:
Sorry for that, again. My editor persistently suggests me format this file. I switched off it.

jessex commented on src/components/controls/RadioGroup.scss:
```diff
@@ -0,0 +1,29 @@
+    top: -3px;
+    *overflow: hidden;
+  }
+}
```
Missing a newline here.

DmitryJakubovsky replied:
Done.

jessex commented on src/components/controls/Select.js:
```diff
@@ -0,0 +1,42 @@
+  isSearchable: false,
+  styles: {
+    container: (styles) => ({ ...styles, ...fontStyles }),
+    // control: (styles) => ({ ...styles, ...fontStyles }),
```
We should remove these comment or use it.

DmitryJakubovsky replied:
Removed it.

DmitryJakubovsky commented:
@jessex 

### Fonts
I think fonts were changed because I added Roboto font (@kbinney asked me about it).
Here you can see that Roboto font is default font, but it was not included before.
https://github.com/Recidiviz/pulse-dashboard/blob/e41e8f4829ea7f49423048522c9a37da7bbb6fb2/src/assets/styles/spec/settings/fonts.scss#L1-L4
After I added this font it applies. Should I disable this font?

### Icons
I checked, an avatar is a dynamic image and it received by the API. `Demo Jones` avatar has dimensions 64x64. 
![avatar](https://ui-avatars.com/api/?name=Demo+Jones&background=0D8ABC&color=fff&rounded=true)
I think your avatar has its own dimensions.

### Geo View

My fail. Fixed.

jessex commented:
You're right about the avatar. Thanks for solving that, @DmitryJakubovsky! And thanks for fixing the map/graph button logic. 

@kbinney, can you weigh in on the fonts here? Essentially, we had a misconfiguration that Dmitry has "solved" but the solution means changing most of the fonts in the app to Roboto. Would you prefer to maintain the fonts we already have? That's the last question here before we can launch this.

kbinney commented:
I would prefer to maintain the fonts we already have, but am confused if that means the explore bar will look like it did prior to iteration. Is it possible to have the explore bar look how it looks in most recent commit, and the rest of the app maintain the fonts and styles they had prior to this change?

jessex commented:
>Is it possible to have the explore bar look how it looks in most recent commit
Yeah that shouldn't be a problem, it just means the Explore bar and the rest of the app will be using different fonts.

@DmitryJakubovsky once we've made that update, to revert the rest of the app to the fonts they were using previously and keep the Explore bar as is, we'll be ready to merge this.
